.github/workflows/fly-deploy.yml

```diff
@@ -0,0 +1,66 @@
+name: fly.io Deploy
+on:
+  workflow_run:
+    workflows: ["fly.io Build"]
+    types:
+      - completed
+
+jobs:
+  deploy:
+    name: Deploy app to fly.io
+    runs-on: ubuntu-latest
+    if: |
+      github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success'
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up flyctl
+        uses: superfly/flyctl-actions/setup-flyctl@master
+        with:
+          version: 0.2.72
+      - name: Download artifact
+        uses: actions/github-script@v7
+        with:
+          script: |
```

Comment:
It would be helpful to include a comment (around here or at the top) with a link to some reference (like https://securitylab.github.com/research/github-actions-preventing-pwn-requests/, if you used that)